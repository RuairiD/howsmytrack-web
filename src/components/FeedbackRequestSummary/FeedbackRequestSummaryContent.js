import React from "react";

import { Row, Col, Typography } from "antd";
import { Div } from "lemon-reset";
import GENRE_OPTIONS from "../../genres";
import type { FeedbackRequestSummaryProps } from "./FeedbackRequestSummary";
import FeedbackRequestSummaryMedia from "./FeedbackRequestSummaryMedia";

type Props = {
    feedbackRequestSummary: FeedbackRequestSummaryProps,
};

/*
 * Component for inline feedback request preview shown on FeedbackGroupsPage and in FeedbackRequestForm
 */
const FeedbackRequestSummaryContent = ({ feedbackRequestSummary }: Props) => (
    <Div>
        <Row gutter={[16, 16]}>
            <Col>
                <FeedbackRequestSummaryMedia
                    mediaUrl={feedbackRequestSummary.mediaUrl}
                    mediaType={feedbackRequestSummary.mediaType}
                />
            </Col>
        </Row>
        <Row gutter={[16, 16]}>
            <Col>
                <Typography.Text strong>Genre: </Typography.Text>
                <Typography.Text>{GENRE_OPTIONS[feedbackRequestSummary.genre]}</Typography.Text>
            </Col>
        </Row>
        {feedbackRequestSummary.mediaUrl && (
            <Row>
                {feedbackRequestSummary.feedbackPrompt && (
                    <Col>
                        <Typography.Text strong>You said: </Typography.Text>
                        <Typography.Text>"{feedbackRequestSummary.feedbackPrompt}"</Typography.Text>
                    </Col>
                )}
                {!feedbackRequestSummary.feedbackPrompt && (
                    <Col>
                        <Typography.Text strong>You did not provide any additional information for feedback.</Typography.Text>
                    </Col>
                )}
            </Row>
        )}
    </Div>
);

export default FeedbackRequestSummaryContent;
