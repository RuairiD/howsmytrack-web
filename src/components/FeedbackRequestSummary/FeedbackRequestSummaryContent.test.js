import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import FeedbackRequestSummaryContent from "./FeedbackRequestSummaryContent";

describe("FeedbackRequestSummaryContent", () => {
    it("renders a FeedbackRequestSummaryMedia and genre for a trackless request", () => {
        const wrapper = shallow(
            <FeedbackRequestSummaryContent
                feedbackRequestSummary={{
                    genre: "genre",
                }}
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders a FeedbackRequestSummaryMedia without a feedback prompt, if one is not provided", () => {
        const wrapper = shallow(
            <FeedbackRequestSummaryContent
                feedbackRequestSummary={{
                    genre: "genre",
                    mediaUrl: "mediaUrl",
                }}
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders a FeedbackRequestSummaryMedia and feedback prompt, if one is provided", () => {
        const wrapper = shallow(
            <FeedbackRequestSummaryContent
                feedbackRequestSummary={{
                    genre: "genre",
                    mediaUrl: "mediaUrl",
                    feedbackPrompt: "feedbackPrompt",
                }}
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
