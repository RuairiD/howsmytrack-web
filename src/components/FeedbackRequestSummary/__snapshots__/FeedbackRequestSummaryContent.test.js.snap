// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackRequestSummaryContent renders a FeedbackRequestSummaryMedia and feedback prompt, if one is provided 1`] = `
<Div
  className=""
>
  <Row
    gutter={
      Array [
        16,
        16,
      ]
    }
  >
    <Col>
      <FeedbackRequestSummaryMedia
        mediaUrl="mediaUrl"
      />
    </Col>
  </Row>
  <Row
    gutter={
      Array [
        16,
        16,
      ]
    }
  >
    <Col>
      <Text
        strong={true}
      >
        Genre: 
      </Text>
      <Text />
    </Col>
  </Row>
  <Row
    gutter={0}
  >
    <Col>
      <Text
        strong={true}
      >
        You said: 
      </Text>
      <Text>
        "
        feedbackPrompt
        "
      </Text>
    </Col>
  </Row>
</Div>
`;

exports[`FeedbackRequestSummaryContent renders a FeedbackRequestSummaryMedia and genre for a trackless request 1`] = `
<Div
  className=""
>
  <Row
    gutter={
      Array [
        16,
        16,
      ]
    }
  >
    <Col>
      <FeedbackRequestSummaryMedia />
    </Col>
  </Row>
  <Row
    gutter={
      Array [
        16,
        16,
      ]
    }
  >
    <Col>
      <Text
        strong={true}
      >
        Genre: 
      </Text>
      <Text />
    </Col>
  </Row>
</Div>
`;

exports[`FeedbackRequestSummaryContent renders a FeedbackRequestSummaryMedia without a feedback prompt, if one is not provided 1`] = `
<Div
  className=""
>
  <Row
    gutter={
      Array [
        16,
        16,
      ]
    }
  >
    <Col>
      <FeedbackRequestSummaryMedia
        mediaUrl="mediaUrl"
      />
    </Col>
  </Row>
  <Row
    gutter={
      Array [
        16,
        16,
      ]
    }
  >
    <Col>
      <Text
        strong={true}
      >
        Genre: 
      </Text>
      <Text />
    </Col>
  </Row>
  <Row
    gutter={0}
  >
    <Col>
      <Text
        strong={true}
      >
        You did not provide any additional information for feedback.
      </Text>
    </Col>
  </Row>
</Div>
`;
